networkx/algorithms/distance_measures.py

```diff
@@ -412,6 +415,10 @@ def harmonic_diameter(G, sp=None):
     G : NetworkX graph
        A graph
 
+    weight : string or None, optional (default=None)
+       The edge attribute that holds the numerical value used as a weight.
+       If None, then each edge has unit weight.
+
```

Comment:
As noted in the linked issue, this should match the parameter description for `single_source_dijkstra_path_length`